src/dnaio/_core.pyx

```diff
@@ -14,7 +14,14 @@ cdef extern from "Python.h":
     bint PyUnicode_IS_COMPACT_ASCII(object o)
     object PyUnicode_New(Py_ssize_t size, Py_UCS4 maxchar)
 
-cdef extern from "ascii_check.h":
+cdef extern from *:
+    """
+    #if defined(USE_SSE2)
```

Comment:
You could use `#if defined(__x86_64__) || _M_X64` directly and then leave `setup.py` unchanged. `__x86_64__` is for GCC and Clang and `_M_X64` is for Visual Studio, see http://jdebp.info/FGA/predefined-macros-processor.html .

Reply:
The advantage of this way is that it makes it also quite easy to test if the non-sse2 version is correct by undefining the USE_SSE2 macro.
I am aware that compiler-specific options exist, but to me that does not look like a particularly stable, reliable and universal API to build on.

Reply:
I agree with your first paragraph. Testing is a valid reason, absolutely, so I’m fine with using that option.

Regarding your other point, I disagree. Querying a documented, pre-defined compiler macro is totally fine and in my opinion not worse than relying on `platform.machine()`, which is at the same level of "universalness", demonstrated by having to check for both `"x86_64"` and `"AMD64"`.

Reply:
isa-l uses this in at least one of their files:
https://github.com/intel/isa-l/blob/ebb78fc99ef7247bd9c16cecf7faa248563ce4f6/erasure_code/ec_highlevel_func.c#L32